### PR TITLE
GridWorld now uses action masking

### PR DESCRIPTION
**Pull request by vincentpierre**
Action masking makes GridWorld learn faster because it does not waste time bumping into walls.
screen shot 2018-08-28 at 1 53 40 pm

### CODE REVIEW

File: unity-environment/Assets/ML-Agents/Examples/GridWorld/Scripts/GridAgent.cs [hunk 1]
```diff
@@ -16,7 +16,26 @@ public override void InitializeAgent()
 
     public override void CollectObservations()
     {
-
+        // Prevents the agent from picking an action that would make it collide with a wall
+        var positionX = (int) transform.position.x;
+        var positionZ = (int) transform.position.z;
+        var maxPosition = academy.gridSize - 1;
+        if (positionX == 0)
```
Comment (mmattar) [inline unity-environment/Assets/ML-Agents/Examples/GridWorld/Scripts/GridAgent.cs]: the constants 0 -> 3 in the conditionals (if position...) and in SetActionMask should be class static variables.

File: MLAgentsSDK/Assets/ML-Agents/Examples/GridWorld/Scripts/GridAgent.cs [hunk 2]
```diff
@@ -27,22 +51,22 @@ public override void AgentAction(float[] vectorAction, string textAction)
 
         // 0 - Forward, 1 - Backward, 2 - Left, 3 - Right
         Vector3 targetPos = transform.position;
```
Comment (mmattar) [inline MLAgentsSDK/Assets/ML-Agents/Examples/GridWorld/Scripts/GridAgent.cs]: delete this comment

File: MLAgentsSDK/Assets/ML-Agents/Examples/GridWorld/Scripts/GridAgent.cs [hunk 3]
```diff
@@ -9,14 +9,38 @@ public class GridAgent : Agent
...
         academy = FindObjectOfType(typeof(GridAcademy)) as GridAcademy;
     }
 
     public override void CollectObservations()
     {
-
+        // Prevents the agent from picking an action that would make it collide with a wall
+        var positionX = (int) transform.position.x;
+        var positionZ = (int) transform.position.z;
+        var maxPosition = academy.gridSize - 1;
```
Comment (mmattar) [inline MLAgentsSDK/Assets/ML-Agents/Examples/GridWorld/Scripts/GridAgent.cs]: can the grid size be changed at runtime? If not, move this to `InitializeAgent`

Comment (vincentpierre) [reply MLAgentsSDK/Assets/ML-Agents/Examples/GridWorld/Scripts/GridAgent.cs]: It can be changed at runtime yes. It is one of the reset paramters.

**Pull request merged**